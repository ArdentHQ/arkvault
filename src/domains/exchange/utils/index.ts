export * from "./assertions";
export * from "./errors";
