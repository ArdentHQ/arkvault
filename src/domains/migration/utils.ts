import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
