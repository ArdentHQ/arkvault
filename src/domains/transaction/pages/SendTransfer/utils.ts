export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
