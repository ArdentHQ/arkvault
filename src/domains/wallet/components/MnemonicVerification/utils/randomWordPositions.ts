import { sample } from "@ardenthq/sdk-helpers";

const randomWordPositions = (length: number): number[] => {
	const positions: number[] = [...Array.from({ length }).keys()];
	const result: number[] = [];

	while (result.length < 3) {
		const randomNumber = sample(positions) + 1;

		if (result.includes(randomNumber)) {
			continue;
		}

		result.push(randomNumber);
	}

	return result.sort((a, b) => a - b);
};

export { randomWordPositions };
