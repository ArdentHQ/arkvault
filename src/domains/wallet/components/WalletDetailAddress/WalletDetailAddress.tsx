import { Contracts } from "@payvo/sdk-profiles";
import React from "react";
import { useTranslation } from "react-i18next";

import { WalletDetail } from "@/domains/wallet/components/WalletDetail";
import { Address } from "@/app/components/Address";
import { Avatar } from "@/app/components/Avatar";

export const WalletDetailAddress = ({ wallet }: { wallet: Contracts.IReadWriteWallet }) => {
	const { t } = useTranslation();

	return (
		<WalletDetail
			label={t("COMMON.ADDRESS")}
			extra={
				<>
					<div className="flex items-center sm:hidden">
						<Avatar size="xs" address={wallet.address()} />
					</div>
					<div className="hidden sm:flex sm:items-center">
						<Avatar size="lg" address={wallet.address()} />
					</div>
				</>
			}
		>
			<Address address={wallet.address()} />
		</WalletDetail>
	);
};
