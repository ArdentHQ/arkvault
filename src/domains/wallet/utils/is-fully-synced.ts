import { Contracts } from "@payvo/sdk-profiles";

export const isFullySynced = (wallet: Contracts.IReadWriteWallet): boolean =>
	wallet.hasBeenFullyRestored() && (wallet.hasSyncedWithNetwork() || wallet.balance() === 0);
