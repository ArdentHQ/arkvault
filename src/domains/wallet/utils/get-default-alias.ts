import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { networkDisplayName } from "@/utils/network-utils";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${networkDisplayName(network)} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
