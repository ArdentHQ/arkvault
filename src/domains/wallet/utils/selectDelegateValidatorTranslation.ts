import { Networks } from "@ardenthq/sdk";
import { isMainsailNetwork } from "@/utils/network-utils";

interface Properties<T extends string | React.ReactNode> {
	network: Networks.Network;
	validatorStr: T;
	delegateStr: T;
}

export const selectDelegateValidatorTranslation = <T extends string | React.ReactNode>({ network, validatorStr, delegateStr }: Properties<T>): T => isMainsailNetwork(network) ? validatorStr : delegateStr;



