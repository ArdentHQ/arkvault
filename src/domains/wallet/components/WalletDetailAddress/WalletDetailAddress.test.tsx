import React from "react";

import { env, getDefaultProfileId, render } from "@/utils/testing-library";

import { WalletDetailAddress } from "./WalletDetailAddress";

describe("WalletDetailAddress", () => {
	it("should render", () => {
		const profile = env.profiles().findById(getDefaultProfileId());

		const { container } = render(<WalletDetailAddress address={profile.wallets().first().address()} />);

		expect(container).toMatchSnapshot();
	});
});
