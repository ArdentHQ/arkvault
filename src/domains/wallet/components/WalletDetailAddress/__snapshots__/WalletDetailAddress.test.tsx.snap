// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WalletDetailAddress > should render 1`] = `
<div>
  <div
    class="no-ligatures css-19rivvh"
    data-testid="TransactionDetail"
  >
    <div
      class="flex w-full items-center justify-between space-x-4"
    >
      <div
        class="w-40 space-y-2 whitespace-nowrap sm:flex-1"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Address
        </div>
        <div
          class="hidden items-center font-semibold sm:flex"
        >
          <div
            class="flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
          >
            <span
              class="no-ligatures min-w-0 overflow-hidden text-theme-text font-semibold text-base"
              data-testid="Address__address"
            >
              D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex w-full items-center justify-end overflow-auto sm:w-auto"
      >
        <div
          class="flex w-full items-center justify-end overflow-auto pr-4 font-semibold sm:hidden"
        >
          <div
            class="w-full truncate"
          >
            <div
              class="flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
            >
              <span
                class="no-ligatures min-w-0 overflow-hidden text-theme-text font-semibold text-base"
                data-testid="Address__address"
              >
                D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex items-center sm:hidden"
        >
          <div
            class="shrink-0 css-1vy2z2u"
            data-testid="Avatar"
          >
            <div
              class="inline-flex h-full w-full items-center justify-center overflow-hidden rounded-full align-middle"
            >
              <img
                alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                src="data:image/svg+xml;utf8,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" class="picasso" width="100" height="100" viewBox="0 0 100 100"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill="rgb(0, 150, 136)" width="100" height="100"/><circle r="50" cx="80" cy="50" fill="rgb(76, 175, 80)"/><circle r="35" cx="100" cy="40" fill="rgb(156, 39, 176)"/><circle r="40" cx="90" cy="70" fill="rgb(3, 169, 244)"/></svg>"
                title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
              />
            </div>
          </div>
        </div>
        <div
          class="hidden sm:flex sm:items-center"
        >
          <div
            class="shrink-0 css-8d0u1i"
            data-testid="Avatar"
          >
            <div
              class="inline-flex h-full w-full items-center justify-center overflow-hidden rounded-full align-middle"
            >
              <img
                alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                src="data:image/svg+xml;utf8,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" class="picasso" width="100" height="100" viewBox="0 0 100 100"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill="rgb(0, 150, 136)" width="100" height="100"/><circle r="50" cx="80" cy="50" fill="rgb(76, 175, 80)"/><circle r="35" cx="100" cy="40" fill="rgb(156, 39, 176)"/><circle r="40" cx="90" cy="70" fill="rgb(3, 169, 244)"/></svg>"
                title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
